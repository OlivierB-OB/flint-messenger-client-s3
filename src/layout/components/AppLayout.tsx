import { makeStyles, createStyles, Theme } from '@material-ui/core/styles';
import React from 'react';
import { connect } from 'react-redux';
import { AppDrawer, drawerWidth } from './AppDrawer';
import { AppMenu } from './AppMenu';
import { AppContent } from './AppContent';
import { IAppState } from '../../appReducer';

interface ILayoutDisplayProps {
  showDrawer: boolean;
}

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    content: {
      width: `100%`,
      transition: theme.transitions.create(['margin', 'width'], {
        easing: theme.transitions.easing.sharp,
        duration: theme.transitions.duration.leavingScreen,
      }),
    },
    contentShift: {
      width: `calc(100% - ${drawerWidth}px)`,
      marginLeft: drawerWidth,
      transition: theme.transitions.create(['margin', 'width'], {
        easing: theme.transitions.easing.easeOut,
        duration: theme.transitions.duration.enteringScreen,
      }),
    },
    drawer: {
      width: drawerWidth,
    },
  }),
);

export function LayoutDisplay({ showDrawer }: ILayoutDisplayProps) {
  const classes = useStyles();
  const contentClasses = [classes.content, showDrawer && classes.contentShift].filter(Boolean).join(' ');
  return (
    <div>
      <div className={contentClasses}>
        <AppMenu />
        <AppContent />
      </div>
      <AppDrawer />
    </div>
  );
}

const mapStateToProps = ({ layout }: IAppState) => ({
  showDrawer: layout.showDrawer,
});

export const AppLayout = connect(mapStateToProps)(LayoutDisplay);
