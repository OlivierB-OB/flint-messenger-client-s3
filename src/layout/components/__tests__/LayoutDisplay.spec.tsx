import React from 'react';
import { shallow } from 'enzyme';
import { LayoutDisplay } from '../AppLayout';

describe('LayoutDisplay', () => {
  it('display without drawer', async () => {
    const component = shallow(<LayoutDisplay showDrawer={false} />);
    expect(component).toMatchSnapshot();
  });

  it('display with drawer', async () => {
    const component = shallow(<LayoutDisplay showDrawer={true} />);
    expect(component).toMatchSnapshot();
  });
});
