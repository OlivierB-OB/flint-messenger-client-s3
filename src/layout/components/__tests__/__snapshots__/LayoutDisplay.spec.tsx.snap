// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LayoutDisplay display with drawer 1`] = `
<div>
  <div
    className="makeStyles-content-1 makeStyles-contentShift-2"
  >
    <Connect(Menu) />
    <AppContent />
  </div>
  <Connect(DrawerDisplay) />
</div>
`;

exports[`LayoutDisplay display without drawer 1`] = `
<div>
  <div
    className="makeStyles-content-1"
  >
    <Connect(Menu) />
    <AppContent />
  </div>
  <Connect(DrawerDisplay) />
</div>
`;
